### DISCUSSION TITLE
Suggestion to change crypto-js to the native crypto

### DISCUSSION THREAD

**Discussion by joaoguidev** [sergiodxa/remix-utils#324] [Ideas]
Hello.

First of all execelent set of tools! 

I'm using the CSRF and it is great. 

I would like to suggest using the native crypto instead of crypto-js for the utils.

It might be an underlying condition for using crypto-js but otherwise it seems a good move.

Thank you.

**Comment by sergiodxa** [comment]
This has been suggested before, Remix Utils targets the same Node versions that Remix supports, and all the runtimes that Remix supports (Node, Cloudflare, Deno and Bun), and because Remix targets Node v18 this can't be done.

In Node v18 you have to import crypto from `node:crypto`, while in other runtimes you have to use `globalThis.crypto`.

Once Remix drops support for Node v18 and target only v20 or greater it will be possible to do this change since in v20 Node now defines `globalThis.crypto` too.

**Reply by joaoguidev** [reply]
Awesome.

Thank you.

I got it.
